io.openems.edge.evcs.api/src/io/openems/edge/evcs/v2/api/EvcsPower.java

```diff
@@ -0,0 +1,20 @@
+package io.openems.edge.evcs.v2.api;
+
+import io.openems.edge.common.filter.RampFilter;
+
+public interface EvcsPower {
```

Comment:
very nice to use a Power model!

Reply:
Ideally I would like to merge the existing EssPower - that does power distribution for energy storage system clusters - with the EvcsPower. Because we eventually need to distribute 'power' (and separately 'energy') among all managed and unmanaged devices.

The EssPower internally uses a linear equation system with linear constraints to model the possibilities and enable power distribution. We had a plan to migrate this linear equation system to OjAlgo, which allows quadratic optimization as well, so that we could write smart(er) target functions for the power distribution than our current manual 'Optimizers' (https://github.com/OpenEMS/openems/tree/develop/io.openems.edge.ess.core/src/io/openems/edge/ess/core/power/optimizers). 

Still... this is a huge, separate topic and should not block the definition of this API.

Reply:
hi @sfeilmeier, the middle plan was also to use linear equations for the "chargecontrolling", because we really loved that approach. yes quadratic would be amazing, than you could 'punish' but still allow a bit 'overcurrent' and get a better result. i did that in my theses, but the 'runtime' was at that time very long. but if it gets an fast 'solution' that would be fantastic.